service/src/main/java/uk/gov/hmcts/reform/fpl/service/InboxLookupService.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.config.GeneralInboxLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityEmailLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import static net.logstash.logback.encoder.org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+@Service
+public class InboxLookupService {
+    private final ObjectMapper objectMapper;
+    private final LocalAuthorityEmailLookupConfiguration localAuthorityEmailLookupConfiguration;
+    private final GeneralInboxLookupConfiguration generalInboxLookupConfiguration;
+
+    public InboxLookupService(ObjectMapper objectMapper,
+                              LocalAuthorityEmailLookupConfiguration localAuthorityEmailLookupConfiguration,
+                              GeneralInboxLookupConfiguration generalInboxLookupConfiguration) {
+        this.objectMapper = objectMapper;
+        this.localAuthorityEmailLookupConfiguration = localAuthorityEmailLookupConfiguration;
+        this.generalInboxLookupConfiguration = generalInboxLookupConfiguration;
+    }
+
+    public String getLocalAuthorityOrFallbackEmail(CaseDetails caseDetails, String localAuthorityCode) {
+        String email = localAuthorityEmailLookupConfiguration.getLocalAuthority(localAuthorityCode).getEmail();
+        if (isEmpty(email)) {
+            CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
```

Comment:
I would perform this mapping in the controller, and then pass maybe the solicitor or just the solicitor email directly. Just means that you aren't passing unnecessary information to the method.